new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleTraefikValidation.java

```diff
@@ -0,0 +1,692 @@
+        String.format("createSecret failed for %s", repoSecretName));
+    assertTrue(secretCreated, String.format("createSecret failed while creating secret %s", repoSecretName));
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    String adminSecretName = "weblogic-credentials";
+    Map<String, String> adminSecretMap = new HashMap();
+    adminSecretMap.put("username", "weblogic");
+    adminSecretMap.put("password", "welcome1");
+    secretCreated = assertDoesNotThrow(() -> createSecret(new V1Secret()
+        .metadata(new V1ObjectMeta()
+            .name(adminSecretName)
+            .namespace(domainNamespace))
+        .stringData(adminSecretMap)), "Create secret failed with ApiException");
+    assertTrue(secretCreated, String.format("create secret failed for %s", adminSecretName));
+
+    // create encryption secret
+    logger.info("Create encryption secret");
+    String encryptionSecretName = "encryptionsecret";
+    Map<String, String> encryptionSecretMap = new HashMap();
+    encryptionSecretMap.put("username", "weblogicenc");
+    encryptionSecretMap.put("password", "weblogicenc");
+    secretCreated = assertDoesNotThrow(() -> createSecret(new V1Secret()
+        .metadata(new V1ObjectMeta()
+            .name(encryptionSecretName)
+            .namespace(domainNamespace))
+        .stringData(encryptionSecretMap)), "Create secret failed with ApiException");
+    assertTrue(secretCreated, String.format("create secret failed for %s", encryptionSecretName));
+
+    // create the domain CR
+    Domain domain = new Domain()
+        .apiVersion(API_VERSION)
+        .kind("Domain")
+        .metadata(new V1ObjectMeta()
+            .name(domainUID)
+            .namespace(domainNamespace))
+        .spec(new DomainSpec()
+            .domainUid(domainUID)
+            .domainHomeSourceType("FromModel")
+            .image(miiImage)
+            .addImagePullSecretsItem(new V1LocalObjectReference()
+                .name(repoSecretName))
+            .webLogicCredentialsSecret(new V1SecretReference()
+                .name(adminSecretName)
+                .namespace(domainNamespace))
+            .includeServerOutInPodLog(true)
+            .serverStartPolicy("IF_NEEDED")
+            .serverPod(new ServerPod()
+                .addEnvItem(new V1EnvVar()
+                    .name("JAVA_OPTIONS")
+                    .value("-Dweblogic.StdoutDebugEnabled=false"))
+                .addEnvItem(new V1EnvVar()
+                    .name("USER_MEM_ARGS")
+                    .value("-Djava.security.egd=file:/dev/./urandom ")))
+            .adminServer(new AdminServer()
+                .serverStartState("RUNNING")
+                .adminService(new AdminService()
+                    .addChannelsItem(new Channel()
+                        .channelName("default")
+                        .nodePort(30711))))
```

Comment:
can we get a free port instead of hard coding ?

Reply:
added a new method to get a free port.